pkg/clients/database/cosmosdb/account.go

```diff
@@ -0,0 +1,231 @@
+	// asouza: only keep attributes that can be modified in the comparison.
+	return (equalConsistencyPolicyIfNotNull(p.ConsistencyPolicy, o.ConsistencyPolicy) &&
+		checkEqualLocations(p.Locations, o.Locations) &&
+		equalBoolIfNotNull(p.EnableAutomaticFailover, o.EnableAutomaticFailover) &&
+		equalBoolIfNotNull(p.EnableMultipleWriteLocations, o.EnableMultipleWriteLocations))
+}
+
+func equalConsistencyPolicyIfNotNull(spec, current *v1beta1.CosmosDBAccountConsistencyPolicy) bool {
+	if spec != nil {
+		return (spec == current) || (*spec == *current)
+	}
+
+	return true
+}
+
+func equalBoolIfNotNull(spec, current *bool) bool {
+	return azure.ToBool(spec) == azure.ToBool(current)
+}
+
+func toDatabaseConsistencyPolicy(a *v1beta1.CosmosDBAccountConsistencyPolicy) *documentdb.ConsistencyPolicy {
+	if a == nil {
+		return nil
+	}
+
+	return &documentdb.ConsistencyPolicy{
+		DefaultConsistencyLevel: documentdb.DefaultConsistencyLevel(a.DefaultConsistencyLevel),
+		MaxStalenessPrefix:      a.MaxStalenessPrefix,
+		MaxIntervalInSeconds:    a.MaxIntervalInSeconds,
+	}
+}
+
+func fromDatabaseConsistencyPolicy(a *documentdb.ConsistencyPolicy) *v1beta1.CosmosDBAccountConsistencyPolicy {
+	if a == nil {
+		return nil
+	}
+
+	return &v1beta1.CosmosDBAccountConsistencyPolicy{
+		DefaultConsistencyLevel: string(a.DefaultConsistencyLevel),
+		MaxStalenessPrefix:      a.MaxStalenessPrefix,
+		MaxIntervalInSeconds:    a.MaxIntervalInSeconds,
+	}
+}
+
+func toDatabaseLocations(a []v1beta1.CosmosDBAccountLocation) *[]documentdb.Location {
+	if a == nil {
+		return &[]documentdb.Location{}
+	}
+
+	s := make([]documentdb.Location, len(a))
+	for i, location := range a {
+		s[i] = documentdb.Location{
+			LocationName:     &location.LocationName,
+			FailoverPriority: &location.FailoverPriority,
+			IsZoneRedundant:  &location.IsZoneRedundant,
+		}
+	}
+
+	return &s
+}
+
+func fromDatabaseLocations(a *[]documentdb.Location) []v1beta1.CosmosDBAccountLocation {
+	lenA := 0
+	if a != nil {
+		lenA = len(*a)
+	}
+	s := make([]v1beta1.CosmosDBAccountLocation, lenA)
+	if lenA > 0 {
+		for i, location := range *a {
+			s[i] = v1beta1.CosmosDBAccountLocation{
+				LocationName:     *location.LocationName,
+				FailoverPriority: *location.FailoverPriority,
+				IsZoneRedundant:  *location.IsZoneRedundant,
+			}
+		}
+	}
+	return s
+}
+
+func sortLocations(a []v1beta1.CosmosDBAccountLocation) {
+	sort.SliceStable(a, func(i, j int) bool {
+		return a[i].LocationName < a[j].LocationName
+	})
+}
+
+func checkEqualLocations(a, b []v1beta1.CosmosDBAccountLocation) bool {
+	sortLocations(a)
+	sortLocations(b)
+
+	if len(a) != len(b) {
+		return false
+	}
+
+	for i := range a {
+		if a[i].LocationName != b[i].LocationName {
+			return false
+		}
+		if a[i].FailoverPriority != b[i].FailoverPriority {
+			return false
+		}
+		if a[i].IsZoneRedundant != b[i].IsZoneRedundant {
+			return false
+		}
+	}
+
+	return true
```

Comment:
Nit: We frequently use `cmp` for this kind of thing, which would make this function something like:

```go
return cmp.Equal(a, b, cmp.SortSlices(func(i, j v1beta1.CosmosDBAccountLocation) bool { return i.LocationName < j.LocationName }))
```